optional_plugins/mail/avocado_job_mail.py

```diff
@@ -0,0 +1,260 @@
+import os
+import smtplib
+import time
+import json
+from email.mime.text import MIMEText
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import Init, JobPost, JobPre
+from avocado.core.settings import settings
+
+
+class MailInit(Init):
+    name = "mail-init"
+    description = "Mail plugin initialization"
+
+    def initialize(self):
+        help_msg = "Mail recipient."
+        settings.register_option(
+            section="plugins.job.mail",
+            key="recipient",
+            default="root@localhost.localdomain",
+            help_msg=help_msg,
+        )
+
+        help_msg = "Mail header."
+        settings.register_option(
+            section="plugins.job.mail",
+            key="header",
+            default="[AVOCADO JOB NOTIFICATION]",
+            help_msg=help_msg,
+        )
+
+        help_msg = "Mail sender."
+        settings.register_option(
+            section="plugins.job.mail",
+            key="sender",
+            default="avocado@localhost.localdomain",
+            help_msg=help_msg,
+        )
+
+        help_msg = "Mail server."
+        settings.register_option(
+            section="plugins.job.mail",
+            key="server",
+            default="localhost",
+            help_msg=help_msg,
+        )
+
+        help_msg = "Mail server port."
+        settings.register_option(
+            section="plugins.job.mail",
+            key="port",
+            default=587,
+            help_msg=help_msg,
+        )
+
+        help_msg = "Mail server Application Password."
+        settings.register_option(
+            section="plugins.job.mail",
+            key="password",
+            default="",
+            help_msg=help_msg,
+        )
+
+        help_msg = "Email detail level (basic or advanced)."
+        settings.register_option(
+            section="plugins.job.mail",
+            key="detail_level",
+            default="basic",
+            help_msg=help_msg,
+        )
+
+
+class Mail(JobPre, JobPost):
+    name = "mail"
+    description = "Sends mail to notify on job start/end"
+
+    @staticmethod
+    def _get_smtp_config(job):
+        return (
+            job.config.get("plugins.job.mail.server"),
+            job.config.get("plugins.job.mail.port"),
+            job.config.get("plugins.job.mail.sender"),
+            job.config.get("plugins.job.mail.password", ""),
+        )
+
+    @staticmethod
+    def _build_message(
+        job, time_content, Type, result_info="", finishedtime="", test_summary=""
```

Comment:
While the lint is not unhappy about this parameter being named type, using a Capitalized variable name is a one of a kind in this repo, and, it doesn't feel very Pythonic.  I'd rename this "phase" or "plugin_type" or something like that.

Reply:
Replaced with "phase"